queue_services/entity-emailer/src/entity_emailer/resources/worker.py

```diff
@@ -0,0 +1,224 @@
+    2. Get email message
+    3. Process email
+    4. Send email
+    5. Publish to event Q
+
+    Decisions on returning a 2xx or failing value to
+    the Queue should be noted here:
+    - Empty or garbaled messages are knocked off the Q
+    - If there is no email to send, knock off Q
+    - If email object(s) are empty or missing, knock off Q
+    - If email failed to send, put back on Q
+    """
+    structured_log(request, "INFO", f"Incoming raw msg: {request.data}")
+    
+    # 1. Get cloud event
+    # ##
+    if not (ce := queue.get_simple_cloud_event(request)):
+        #
+        # Decision here is to return a 200,
+        # so the event is removed from the Queue
+        return {}, HTTPStatus.OK
+
+    structured_log(request, "INFO", f"received ce: {str(ce)}")
+
+    # 2. Get email message
+    # ##     
+    if not (email_msg := json.loads(ce.data.decode('utf-8'))):
+      # no email message, take off queue
+      return {}, HTTPStatus.OK
+    
+    structured_log(request, 'INFO', f'Extracted email msg: {email_msg}')
+
+    # 3. Process email
+    # ##
+    token = AccountService.get_bearer_token()
+    if not (email := process_email(email_msg, token)):
+        # no email to send, take off queue
+        structured_log(request, 'INFO', f'No email to send for: {email_msg}')
+        return {}, HTTPStatus.OK
+    
+    # 4. Send email
+    # ##
+    if not email \
+            or 'recipients' not in email \
+            or 'content' not in email \
+            or 'body' not in email['content']:
+        # email object(s) is empty, take off queue
+        structured_log(request, 'INFO', 'Send email: email object(s) is empty')
+        return {}, HTTPStatus.OK
+
+    if not email['recipients'] \
+            or not email['content'] \
+            or not email['content']['body']:
+        # email object(s) is missing, take off queue
+        structured_log(request, 'INFO', 'Send email: email object(s) is missing')
+        return {}, HTTPStatus.OK
+
+    resp = send_email(email, token)
+    
+    if resp.status_code != HTTPStatus.OK:
+        # log the error and put the email msg back on the queue
+        structured_log(request, 'ERROR', f'Queue Error - email failed to send: {json.dumps(email_msg)}'
+                      '\n\nThis message has been put back on the queue for reprocessing.')
+        return {}, HTTPStatus.NOT_FOUND
+    
+    # 5. Publish to event Q
+    # ##
+    cloud_event = SimpleCloudEvent(
+          source=__name__[: __name__.find(".")],
+          subject="email",
+          type="Email",
+          data=email
+    )
+    try:
+        event_topic = current_app.config.get("ENTITY_EVENT_TOPIC", "events")
+        queue.publish(topic=event_topic, payload=queue.to_queue_message(cloud_event))
+    except Exception as err:  # noqa B902; pylint: disable=W0703; we don't want to fail out the email, so ignore all.
+        structured_log(request, 'ERROR', f'Queue Publish Event Error: err={err} email={email}')
```

Comment:
I don't think we need to publish  to any queues from the emailer.  I know we had the `publish_event` function that published to the "ENTITY_EVENT_TOPIC" topic but it isnever used. 

I think we can remove this.

Reply:
Do we want to keep the GCP Queue code after removing? This is the only place that something is published to a queue.

Reply:
Leave this code for now.  Maybe @thorwolpert or @vysakh-menon-aot have background on whether we eventually plan on publishing to the event topic or why it was there in the first place.

Reply:
We can keep the GCP queue service code.  

I took a deeper look at worker.py code and we need it even if we don't publish a queue message.  I believe it's used at least to initialize the queue and read incoming queue messages.